ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpService.java

```diff
@@ -229,12 +233,20 @@ private void validateConfig(final JsonRpcConfiguration config) {
                 LOG.info(
                     "JsonRPC service started and listening on {}:{}", config.getHost(), actualPort);
                 config.setPort(actualPort);
-                // Request that a NAT port forward for our server port
-                if (natManager.isPresent()) {
-                  natManager
-                      .get()
-                      .requestPortForward(
-                          config.getPort(), UpnpNatManager.Protocol.TCP, "partheon-json-rpc");
+
+                if (natService.isPresent()) {
+                  final NatService service = natService.get();
+                  // request that a NAT port forward for our server port (if UPNP Nat Environment)
+                  if (service.isNatEnvironment() && service.getNatMethod().equals(NatMethod.UPNP)) {
+                    service
+                        .getNatManager()
+                        .ifPresent(
+                            natSystem -> {
+                              final UpnpNatManager upnpNatSystem = (UpnpNatManager) natSystem;
```

Comment:
What do you think about having things like `requestPortForward` be on the interface and other managers can throw `NotSupportedException`s? I'm not saying that's better but it does avoid a cast here. Maybe the cast is actually the more elegant option. Thoughts?

Reply:
that's what I wanted to do at the beginning but Danno preferred that we remove these methods from the generic API because it's too specific to UPNP

Reply:
Makes sense, just wanted to get your thoughts.